# Assignment 01: Part-of-Speech (POS) Extraction from a News Article

**Name:** Yominesh Giri

**Objective:** Apply Natural Language Processing techniques to identify and extract nouns and verbs from an English news article.

**Article source:** BBC News – "How Wimbledon cracked cricket-obsessed India"

In [1]:
!pip install spacy
!python -m spacy download en_core_web_sm

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.8/12.8 MB 81.8 MB/s eta 0:00:00
✔ Download and installation successful
You can now load the package via spacy.load('en_core_web_sm')
⚠ Restart to reload dependencies
If you are in a Jupyter or Colab notebook, you may need to restart Python in
order to load all the package's dependencies. You can do this by selecting the
'Restart kernel' or 'Restart runtime' option.


## Load the spaCy Model and the Article

Load the pretrained English model into memory, then read the article text from `article.txt`.

In [6]:
import spacy

nlp = spacy.load("en_core_web_sm")
print("Model loaded successfully.")

with open("article.txt", "r", encoding="utf-8") as f:
    text = f.read()

print(f"Loaded article with {len(text.split())} words.\n")
print(text[:300], "...")

Model loaded successfully.
Loaded article with 863 words.

Few things are more sacrosanct in sport than Wimbledon.

Its all-white dress code for players, minimalist court-side advertising and association with strawberries and cream all remain fiercely guarded traditions.

Yet as the organisers of the world's oldest Grand Slam set out to crack cricket-obsess ...


## Perform POS Tagging

Running the text through the spaCy pipeline tokenizes it and assigns a POS tag to every single token.

In [7]:
doc = nlp(text)

print("Sample of POS-tagged tokens:\n")
for token in list(doc)[:20]:
    print(f"{token.text:<15} {token.pos_}")

Sample of POS-tagged tokens:

Few             ADJ
things          NOUN
are             AUX
more            ADV
sacrosanct      ADJ
in              ADP
sport           NOUN
than            ADP
Wimbledon       PROPN
.               PUNCT


              SPACE
Its             PRON
all             ADV
-               PUNCT
white           ADJ
dress           NOUN
code            NOUN
for             ADP
players         NOUN
,               PUNCT


## Extract Nouns and Verbs

Only keep tokens tagged NOUN or VERB. Token needs to be alphabetic, which filters out punctuation, numbers, and whitespace automatically.

In [8]:
target_pos = {"NOUN", "VERB"}

extracted_words = [
    (token.text, token.pos_)
    for token in doc
    if token.pos_ in target_pos and token.is_alpha
]

print(f"Total nouns and verbs extracted: {len(extracted_words)}")
print("\nFirst 15 extracted words:")
for word, pos in extracted_words[:15]:
    print(f"{word:<15} {pos}")

Total nouns and verbs extracted: 331

First 15 extracted words:
things          NOUN
sport           NOUN
dress           NOUN
code            NOUN
players         NOUN
minimalist      NOUN
court           NOUN
side            NOUN
advertising     NOUN
association     NOUN
strawberries    NOUN
cream           NOUN
remain          VERB
guarded         VERB
traditions      NOUN


## Save Extracted Words to CSV

In [9]:
import csv

OUTPUT_CSV = "YomineshGiri_POS_01.csv"

with open(OUTPUT_CSV, "w", newline="", encoding="utf-8") as f:
    writer = csv.writer(f)
    writer.writerow(["Word", "POS_Tag"])
    writer.writerows(extracted_words)

print(f"Saved {len(extracted_words)} rows to {OUTPUT_CSV}")

Saved 331 rows to YomineshGiri_POS_01.csv


## Conclusion

For this assignment, a BBC News article titled "How Wimbledon cracked cricket-obsessed India" was used and processed through spaCy's POS tagger. After loading the article as a text file, the pretrained `en_core_web_sm` model was used to tag every word in the article with its part of speech.

Once every word had a tag the results were filtered to keep only nouns and verbs. Out of 863 words in the article, 331 were extracted as nouns or verbs and saved into a CSV file with two columns: Word and POS_Tag.

Overall, this task demonstrated how POS tagging works in practice i.e. the model doesn't just look words up in a list, it predicts each tag based on the context of the sentence which is why the same word can be tagged differently depending on how it's used.